girder/models/folder.py

```diff
@@ -735,8 +735,11 @@ def copyFolder(self, srcFolder, parent=None, name=None, description=None,
             name = srcFolder['name']
         if description is None:
             description = srcFolder['description']
-        if public == 'original':
-            public = srcFolder.get('public', None)
+        if public is not None and isinstance(public, six.string_types):
+            if public == 'original':
+                public = srcFolder.get('public', None)
+            else:
+                public = self.boolParam('public', params)
```

Comment:
Sorry, forgot this was an outdated method since autoDescribeRoute. We can change to

    public = public == 'true'